test/gmp.jl

```diff
@@ -815,4 +815,10 @@ end
         bfloat = big(11.0)^i
         @test (hash(bint) == hash(bfloat)) == (bint == bfloat)
     end
+    for n in [-12345, -12, -1, 0, 1, 17, 2049]
+        # hashes of small numbers should coincide with Int
+        @test hash(n) == hash(big(n))
+        # Base.hash_integer should coincide with hash for small numbers (#58386)
```

Comment:
Would be good to define "small number" so that we have a clear policy moving forward